src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+            }
+            else {
+                mNextMessageNo = 0x607;
+            }
+        }
+    }
+    else if (dComIfGs_isEventBit(0xA08)) {
+        if (dComIfGs_isEventBit(0x2A02)) {
+            mNextMessageNo = 0xd5e;
+        }
+        else {
+            if (dComIfGs_isEventBit(0xA04)) {
+                mNextMessageNo = 0x5f9;
+            }
+            else {
+                mNextMessageNo = 0x5f8;
+            }
+        }
+    }
+    else if (dComIfGs_isEventBit(0xA20)) {
+        if (dComIfGs_isSymbol(2)) {
+            mNextMessageNo = 0x5f6;
+        }
+        else {
+            if (dComIfGs_isEventBit(0x2B80)) {
+                mNextMessageNo = 0x5f5;
+            }
+            else {
+                mNextMessageNo = 0x5f4;
+            }
+        }
+    }
+    else if (dComIfGs_isEventBit(0xA80)) {
+        if (dComIfGs_isEventBit(0x1980)) {
+            mNextMessageNo = 0xd5d;
+        }
+        else {
+            cXyz* windVec = dKyw_get_wind_vec();
+            if (abs(cM_atan2s(windVec->x, windVec->z)) < 0x1000) {
+                mNextMessageNo = 0x5ee;
+            }
+            else {
+                mNextMessageNo = 0x5ef;
+            }
+        }
+    }
+    else if (dComIfGs_isEventBit(0x902)) {
+        if (dComIfGs_isSymbol(1)) {
+            mNextMessageNo = 0x5ec;
+        }
+        else {
+            if (dComIfGs_isEventBit(0xA10)) {
+                mNextMessageNo = 0x5eb;
+            }
+            else {
+                mNextMessageNo = 0x5ea;
+            }
+        }
+    }
+    else {
+        if (dComIfGs_isEventBit(0x2A08)) {
+            mNextMessageNo = 0xd5c;
+        }
+        else {
+            if (dComIfGs_isEventBit(0x908)) {
+                mNextMessageNo = 0x5df;
+            }
+            else {
+                mNextMessageNo = 0x5e0;
+            }
+        }
+    }
+    return;
 }
 
 /* 00001734-00001BD0       .text setNextMessage__8daShip_cFP9msg_class */
-void daShip_c::setNextMessage(msg_class*) {
-    /* Nonmatching */
+BOOL daShip_c::setNextMessage(msg_class* msg) {
+    uint currMessageID;
+    
+    currMessageID = mNextMessageNo;
+    if ((((((currMessageID != 0x5dd) && (currMessageID != 0x5de)) && (currMessageID != 0x5df)) &&
+         (((currMessageID != 0x5eb && (currMessageID != 0x5ee)) &&
+          ((currMessageID != 0x5ef && ((currMessageID != 0x5f5 && (currMessageID != 0x5f9)))))))) &&
+        ((currMessageID != 0x608 &&
+         ((((((currMessageID != 0x609 && (currMessageID != 0x60a)) && (currMessageID != 0x621)) &&
+            ((currMessageID != 0x623 && (currMessageID != 0x636)))) &&
+           ((currMessageID < 0xd5c || (currMessageID > 0xd78)))) &&
+          ((currMessageID != 0xd7b && (currMessageID != 0x164c)))))))) &&
+       ((currMessageID != 0x1650 &&
+        ((((currMessageID != 0x1651 && (currMessageID != 0x1684)) && (currMessageID != 0x1685)) &&
+         ((currMessageID != 0x1687 && (currMessageID != 0x1689)))))))) {
```

Comment:
Most of these parentheses seem unnecessary, I think you can remove all of them except the pair around the OR condition